### PR TITLE
Development

### PR DESCRIPTION
**Pull request by SebastianPetak**
Hey @KasaiMagi. This is what this app looks like so far.
recipebox
I want to add a feature where you can click on a recipe and something like this will pop up:
screenshot from 2017-03-28 16-32-14
I was thinking of accomplishing this with some sort of modal box for each recipe. Do you think each recipes modal box should be rendered on the main page, or should I figure out a way to render the individual modal box when the user clicks a recipe.

### CODE REVIEW

**Comment by tdbishop** [comment]
nice! Modals are fun - I've done it both ways you thought of. It might be easiest to always have the modal component on the page and then just populate/enable it when you need.

Personal thoughts: I hate modals :P I think they are a UX patch that existed before browsers and JS had the power they have now. Now with single page apps we can dynamically swap contents in and out and provide a better and more seamless experience. You could pull off something similar using react-router by swapping out the visible content and providing an easy way to go 'back'. Technically you could do both since the inner component of the modal would be the same as the inner content of the new route :)

I'm not saying you should stop the modal work, because we all have to do it at some point. But I figured I'd share my opinions about it :)

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
   "homepage": "https://github.com/SebastianPetak/recipe-box-sp#readme",
   "dependencies": {
     "express": "^4.15.2",
+    "lodash": "^4.17.4",
```
Comment (tdbishop) [inline package.json:23]: yus lodash!

File: app/containers/Recipes.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+module.exports = class Recipes extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			recipes: [],
+			localStorageSupport: false
+		};
+		const self = this;
+
+		self.getLocalStorage = () => {
```
Comment (tdbishop) [inline app/containers/Recipes.js:22]: I would try to abstract out the `get`ing of data from localstorage to the `setState` that affects this component. Similar to splitting the db connections/query with the sending db data to another place in code.

File: app/js/shortID.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module.exports = function generate(str) {
+	return (
+		str.concat(String(Math.floor(Math.random() * ((9999999 - 9) + 9)))));
```
Comment (tdbishop) [inline app/js/shortID.js:3]: whoa fun, where is this being used though?

Comment (SebastianPetak) [reply app/js/shortID.js:3]: hehe, I was a bit unsure about using this. This is going to generate a unique key when users create their own recipe to ensure the mapped recipes have their own unique key value.

Comment (tdbishop) [reply app/js/shortID.js:3]: yeah creating UUIDs in JS is tough. What you have here will probably work just fine.
There are npm packages for node that might work in browser - check out https://www.npmjs.com/package/uuid although the Issues queue talks about issues with babel - shrug.

**Comment by SebastianPetak** [comment]
Thank you! I wanted to hear what you thought about it.

**Pull request merged**